src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -344,7 +335,17 @@ private function createResponse(MountedComponent $mounted): Response
 
     private function isLiveComponentRequest(Request $request): bool
     {
-        return $request->attributes->has('_live_component');
+        if (!$request->attributes->has('_live_component')) {
+            return false;
+        }
+
+        if ($this->testMode) {
+            return true;
+        }
```

Comment:
This feels a bit weird to have this here... don't you think ?

Reply:
I don't :) It's the way to make the new behavior seamless for existing test suites. The alternative (requiring test cases to all add HTTP_ACCEPT to their calls) looks like forcing boilerplate.

Reply:
In general, not a big fan of having test-or-dev code in the final classes. 

To be honest, I had the _feeling_ this was one of the "untold conventions" in the symfony codebase.. but if you're good with it, i'm good with it :)

Reply:
Suggestion here, what about renaming something like "controlHeader" or even "acceptHeader" instead ? And set it's default value to "true", instead of having a "false" in the constructor ?

Reply:
testMode defaults to true... :)

Reply:
If you mean changing the default behavior, this makes the tests all red - aka it's a BC break.

Reply:
👍